rusk/tests/services/protected.rs

```diff
@@ -0,0 +1,201 @@
+
+const GAS_LIMIT: u64 = 200_000_000;
+
+fn initial_state<P: AsRef<Path>>(dir: P) -> Result<Rusk> {
+    let snapshot = toml::from_str(include_str!("../config/protected.toml"))
+        .expect("Cannot deserialize config");
+
+    new_state(dir, &snapshot)
+}
+
+const SENDER_INDEX: u64 = 0;
+
+fn verify_protected_method(
+    contract: ContractId,
+    protected_method: &str,
+    rusk: &Rusk,
+    wallet: &wallet::Wallet<TestStore, TestStateClient, TestProverClient>,
+    should_discard: bool,
+) {
+    let mut rng = StdRng::seed_from_u64(0xcafe);
+
+    let refund = wallet
+        .public_spend_key(SENDER_INDEX)
+        .expect("Getting a public spend key should succeed");
+
+    let tx = wallet
+        .execute(
+            &mut rng,
+            contract.to_bytes().into(),
+            String::from(protected_method),
+            (),
+            SENDER_INDEX,
+            &refund,
+            GAS_LIMIT,
+            1,
+        )
+        .expect("Making the transaction should succeed");
+
+    let expected = if should_discard {
+        Some(ExecuteResult {
+            discarded: 1,
+            executed: 0,
+        })
+    } else {
+        None
+    };
+
+    let spent_transactions = generator_procedure(
+        rusk,
+        &[tx],
+        BLOCK_HEIGHT,
+        BLOCK_GAS_LIMIT,
+        vec![],
+        expected,
+    )
+    .expect("generator procedure should succeed");
+
+    if !should_discard {
+        assert!(spent_transactions
+            .first()
+            .expect("Transaction should exist")
+            .err
+            .is_some());
+    }
+}
+
+const TRANSFER_DATA_PROTECTED_METHODS: &'static [&'static str] = &[
+    "root",
+    "num_notes",
+    "module_balance",
+    "message",
+    "opening",
+    "existing_nullifiers",
+    "any_nullifier_exists",
+    "extend_nullifiers",
+    "take_message_from_address_key",
+    "root_exists",
+    "push_message",
+    "take_crossover",
+    "set_crossover",
+    "get_crossover",
+    "extend_notes",
+    "sub_balance",
+    "leaves_from_height",
+    "leaves_from_pos",
+    "push_note",
+    "get_note",
+    "update_root",
+    "add_module_balance",
+    "get_module_balance",
+];
+
+const TRANSFER_LOGIC_PROTECTED_METHODS: &'static [&'static str] = &[
+    "mint",
+    "stct",
+    "wfct",
+    "wfct_raw",
+    "stco",
+    "wfco",
+    "wfco_raw",
+    "wfctc",
+    "root",
+    "num_notes",
+    "module_balance",
+    "message",
+    "opening",
+    "existing_nullifiers",
+    "leaves_from_height",
+    "leaves_from_pos",
+    "spend_and_execute",
+    "refund",
+    "push_note",
+    "update_root",
+    "add_module_balance",
+    "sub_module_balance",
+];
+
+const TRANSFER_PROXY_PROTECTED_METHODS: &'static [&'static str] = &[
+    "spend_and_execute",
+    "refund",
+    "push_note",
+    "update_root",
+    "add_module_balance",
+];
+
+const SHOULD_DISCARD: bool = true;
+const SHOULD_NOT_DISCARD: bool = false; // meaning: should err
```

Comment:
No. Transaction should never be discarded. 
They must be accepted in a block, panic while executing the ICC and result in error

Reply:
removed discarding of transactions